### PR TITLE
Remove most required arguments from electrodes table

### LINKED ISSUE #1442: Remove most required properties from `add_electrode`
This is linked to https://github.com/NeurodataWithoutBorders/nwb-schema/issues/505.

The idea is to remove all the required properties but group and location from the `add_electrode` function:

- x
- y
- z
- imp
- location
- filtering
- group

The logic and background of the request is described in the linked issue above.

### PR DESCRIPTION
**Pull request by h-mayorquin**
## Motivation
Fix #1442 by modifying  the `add_electrode` function in `pynwb.file.py` 

Changes are made to both `file.py` and the corresponding testing modules to drop the strong requirements for x, y, z, imp, and filtering.

Remaining issues:
* There are some roundtrip tests failing. I need to look deeper into this.
* I don't remember if we wanted to add the `allow_positional` argument in the docval of this function, should this be done, @rly? 
* Check corresponding updates in the documentation.

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [X] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured the PR clearly describes the problem and the solution?
- [X] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [X] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [X] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: tests/unit/test_ecephys.py [hunk 1]
```diff
@@ -15,14 +15,11 @@ def make_electrode_table():
...
-    table.add_row(id=2, x=1.0, y=2.0, z=3.0, imp=-2.0, location='CA1', filtering='none',
-                  group=group, group_name='tetrode1')
-    table.add_row(id=3, x=1.0, y=2.0, z=3.0, imp=-3.0, location='CA1', filtering='none',
-                  group=group, group_name='tetrode1')
-    table.add_row(id=4, x=1.0, y=2.0, z=3.0, imp=-4.0, location='CA1', filtering='none',
-                  group=group, group_name='tetrode1')
+    table.add_row(id=1, location='CA1', group=group, group_name='tetrode1')
+    table.add_row(id=1, location='CA1', group=group, group_name='tetrode1')
+    table.add_row(id=1, location='CA1', group=group, group_name='tetrode1')
+    table.add_row(id=1, location='CA1', group=group, group_name='tetrode1')
```
Comment (bendichter) [inline tests/unit/test_ecephys.py]: `id` values, if supplied, should be unique. Also, can you keep this test and add another for the missing values? We want to make sure that supplying x, y, z, etc. continues to work as it did before as long as kwargs are used.

**Comment by rly** [comment]
related: https://github.com/NeurodataWithoutBorders/nwb-schema/pull/506

File: tests/unit/test_ecephys.py (hunk 1 above)
Comment (h-mayorquin) [reply tests/unit/test_ecephys.py]: Ids are not enforced to be unique for the table as defined in that test. As far as I understand, unique ids are enforced in the `add_electrode` function by means of docval decorator's argument `enforce_unique_id`. This is not enforced for the table above which is called with the contructor `ElectrodeTable()`. The latter is just a wrapper for dynamic table creation pre-filled with certain columns:

https://github.com/NeurodataWithoutBorders/pynwb/blob/812275f5be3f73b3d3c358e7509e95b5c01c1e48/src/pynwb/file.py#L1064-L1072

For the second, point, there is already a test passing now where the arguments were not removed:
https://github.com/NeurodataWithoutBorders/pynwb/blob/a7a6e3766219dee55e0ab17110850a14e0c2c274/tests/unit/test_file.py#L244-L268

**Comment by h-mayorquin** [comment]
I am getting the following issue:

```bash
  File "/home/heberto/development/pynwb/.tox/py39/lib/python3.9/site-packages/hdmf/build/objectmapper.py", line 552, in get_attr_value
    raise ContainerConfigurationError(msg)
hdmf.build.errors.ContainerConfigurationError: TimeSeries 'test_ts' does not have attribute 'data__offset' for mapping to spec: {'doc': "Scalar to add to the data after scaling by 'conversion' to finalize its coercion to the specified 'unit'. Two common examples of this include (a) data stored in an unsigned type that requires a shift after scaling to re-center the data, and (b) specialized recording devices that naturally cause a scalar offset with respect to the true units.", 'name': 'offset', 'required': False, 'dtype': 'float32', 'default_value': 0.0}
```
I think this is because the latest version of the schema already has the following changes:
https://github.com/NeurodataWithoutBorders/nwb-schema/pull/494

But the modifications in `pynwb` are not yet implemented:
https://github.com/NeurodataWithoutBorders/pynwb/pull/1424

I created another branch with the changes that I made to the schema (pre addition of offset) and I am testing from there. Do you recommend any other workflow in these situations or do you recommend waiting for that Cody's PR?

Also, is it possible that someone approves the testing workflow in:
https://github.com/NeurodataWithoutBorders/nwb-schema/pull/506
?
I want to see if it is running correctly to assure myself that I am testing my changes here against a valid version of the schema

File: tests/unit/test_ecephys.py (hunk 1 above)
Comment (bendichter) [inline tests/unit/test_ecephys.py]:
```suggestion
    table.add_row(location='CA1', group=group, group_name='tetrode1')
    table.add_row(location='CA1', group=group, group_name='tetrode1')
    table.add_row(location='CA1', group=group, group_name='tetrode1')
    table.add_row(location='CA1', group=group, group_name='tetrode1')
```

**Review by bendichter — APPROVED** [review]

**Review by rly — APPROVED** [review]

**Pull request merged**